packages/starlight/utils/user-config.ts

```diff
@@ -209,15 +209,31 @@ const UserConfigSchema = z.object({
 	/** Disable Starlight's default 404 page. */
 	disable404Route: z.boolean().default(false).describe("Disable Starlight's default 404 page."),
 
+	/**
+	 * Define whether Starlight pages should be prerendered or not.
+	 * Defaults to always prerender Starlight pages, even when the project is
+	 * set to "server" output mode.
+	 * Does nothing when Astro's output mode is undefined or set to "static".
```

Comment:
We throw an error in if you set it to false for static output right? I might just remove this line:
```suggestion
```